src/kedro_templar/core/utils.py

```diff
@@ -0,0 +1,68 @@
+import yaml
+import s3fs
+import logging
+import os
+from pathlib import Path
+from typing import Text, Dict
+from ..import settings
+
+log = logging.getLogger(__name__)
+
+
+def check_folder_structure():
+    """on every run check if it fits to directories from settings"""
+    directories = [settings.TEMPLATES_DIR,
+                   settings.OUTPUT_DIR,
+                   settings.CONFIG_OUTPUT_DIR]
+    for dir in directories:
+        if not os.path.exists(dir):
+            os.makedirs(dir)
+
+def load_config(config_path: Text) -> Dict:
+    """Loads and parases a yaml config into a dict object."""
+    return yaml.full_load(open(config_path))
+
+
+def save_result(result: Text, output_filepath: Path, replace: bool = True):
+    """
+    Saves rendered config into an output file.
+
+    :param result: a rendered template in a text format
+    :param output_filepath: a filepath where the output will be saved
+    :param replace: if set to False, it will only replace the
+                    existing file if it doesn't already exist.
+    """
+    if not replace and output_filepath.exists():
+        log.error(f"Skipping saving a file. {output_filepath} already exists!")
+        return
+
+    if not os.path.exists(output_filepath.parent):
+        log.info(f"Parent directory not found. Creating {output_filepath.parent}.")
+        os.makedirs(output_filepath.parent)
+
+    with open(output_filepath, "w") as output:
+        if type(result)==str:
```

Comment:
`isinstance(result, str)` is a prefered way for checking types in python
or `type(result) is str`

note: `isinstance` is checking all subclass types, while `is` checks only specified type without subclasses.

Reply:
sure